pkg/action/executor/k8s_controller.go

```diff
@@ -176,3 +169,67 @@ func (c *parentController) shouldSkipOperator(controller *unstructured.Unstructu
 func (rc *parentController) String() string {
 	return rc.name
 }
+
+//To determine if ORM should be used or not in terms of the owner reference of the work load controller
+func (c *parentController) shouldUseOrm(controller *unstructured.Unstructured) bool {
+	if controller == nil {
```

Comment:
I don't understand the point of this check. This is checking if `c.obj` is set. We have reached here after doing lot of processing on `c.obj`, isnt it?

Reply:
Did you intend to pass `ownerReferences` as the argument and check `len(controllerOwnerReferences)` and `controllerOwnerReferences[0].Controller != nil` here?

Reply:
> I don't understand the point of this check. This is checking if `c.obj` is set. We have reached here after doing lot of processing on `c.obj`, isnt it?

This check is just for double protection to avoid crash,  @pankajbatra1  also wants me to add this in the function `shouldSkipOperator`  https://github.com/turbonomic/kubeturbo/pull/674/files#r863798437

Reply:
> Did you intend to pass `ownerReferences` as the argument and check `len(controllerOwnerReferences)` and `controllerOwnerReferences[0].Controller != nil` here?

Yes,  we need to check this `Controller ` and `Kind` field in the `ownerRerence` to dertermin if we should ORM to update the workload controller

Reply:
@kevinwangcn You did not understand my point. You are passing and checking the object itself here, which:
1. You don't need to pass as its already available as `c.obj`
2. You don't need to check here, as it has been used extensively before. If its nil the crash will happen earlier but not here. Further we already check for errors getting this object before [here](https://github.com/turbonomic/kubeturbo/blob/master/pkg/action/executor/k8s_controller.go#L64), so I think the nil check on `obj` is not required.
3. What you should ideally check for nil here is the `ownerReferences` (`c.obj.GetOwnerReferences()`)